source/api-abi-versioning/api-versioning.txt

```diff
@@ -0,0 +1,135 @@
+*private API*. For this system to work, you first need to declare a public API.
+This may consist of documentation or be enforced by the code itself. Regardless,
+it is important that this API be clear and precise. Once you identify your
+public API, you communicate changes to it with specific increments to your
+version number.
+
+The API version number is updated according to SemVer and the definition of the public API below.
+
+.. important::
+
+   Only the stability of the public API is communicated by the API version
+   number. Backward (in)compatible changes to the private API are not
+   communicated by the API version number.
+
+.. _cpp-api-source-compatibility:
+
+Source Compatibility
+--------------------
+
+A *public header file* is installed to the include directory for any given CMake build configuration. Any other header file is a *private header file*.
+
+A library's *root namespace* is declared in the global namespace scope, e.g. ``bsoncxx``.
+All entities owned by the library are declared within the library's root namespace.
+Preprocessor macros owned by the library are prefixed with the library name instead, e.g. ``BSONCXX_EXAMPLE_MACRO``.
+
+The *public API* is the set of *documented* entities that are declared or defined in a public header file within the library's root namespace, with the following exceptions:
+
+- The entity is explicitly documented as private or not intended for public use.
+- The entity is declared in a ``detail`` namespace.
+
+Including these exceptions, all other entities are considered to be part of the *private API*.
+
+If there is an entity that *should* be documented, but is currently not, please submit a bug report.
+
+.. important::
+
+   The *documented* behavior of a public entity is also considered part of the
+   public API. Behavior that is not documented is considered library-level
+   undefined behavior and is not supported. If there is behavior that *should*
+   be documented but is currently not, please submit a bug report.
+
+.. note::
+
+   Some entities that are part of the public API may not be part of the stable ABI. Conversely, some symbols that are part of the stable ABI may not be part of the public API. See `ABI Versioning Policy <{+api+}/abi-versioning#binary-compatibility>`__.
+
+.. _cpp-api-build-sys-compat:
+
+Build System Compatibility
+--------------------------
+
+Although the public API depends on the configuration of the build system, the build system itself is **NOT** part of the public API.
+
+Properties of the build system include:
+
+- CMake configuration variables and corresponding behavior (with some exceptions: see below).
+- The name, location, and contents of:
+  - CMake source files (e.g. under ``${PROJECT_SOURCE_DIR}``).
+  - CMake binary files (e.g. under ``${PROJECT_BINARY_DIR}``).
+  - Installed library files (e.g. under ``{CMAKE_INSTALL_LIBDIR}``).
+  - Installed package files (e.g. under ``{CMAKE_INSTALL_LIBDIR}/pkgconfig``).
```

Comment:
This is formatted incorrectly. The list items are not being nested properly:

---

- The name, location, and contents of:
    - CMake source files ...
    - CMake binary files ...
    - Installed package files ...